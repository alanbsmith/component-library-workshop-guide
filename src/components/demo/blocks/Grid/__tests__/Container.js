import React from "react";
import Container from "../Container";
import "jest-styled-components";

import { renderWithTheme } from "__tests__/helpers";

describe("Grid.Container Element", () => {
  it("renders correctly", () => {
    const tree = renderWithTheme(<Container />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("applies the correct style with the `fluid` modifier", () => {
    const tree = renderWithTheme(<Container modifiers={['fluid']} />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule("padding", "0");
  });
});
