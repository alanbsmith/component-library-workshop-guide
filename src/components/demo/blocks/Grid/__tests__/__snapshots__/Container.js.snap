// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid.Container Element applies the correct style with the \`fluid\` modifier 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0.25rem;
  padding: 0;
}

<div
  className="c0"
/>
`;

exports[`Grid.Container Element renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0.25rem;
}

<div
  className="c0"
/>
`;
